Casks/chef-workstation.rb

```diff
@@ -1,20 +1,28 @@
 cask "chef-workstation" do
-  version "22.1.745"
-  sha256 "f6beaa7b00ecb062d541648e6562c9dba3c5380176c2cef612518b114cee712a"
+  arch = Hardware::CPU.intel? ? "x86_64" : "arm64"
+  macos_version = Hardware::CPU.intel? ? "10.15" : "11"
```

Comment:
How come we're changing the MacOS version depending on the CPU arch?

Reply:
We provide the same binary for Intel macOS versions 10.15 and above, this is reference to the base version. M1 macOS versions start from 11 so that's the base version specified. The packages.chef.io URL is differentiated based on both architecture and macOS version. Eventually the macOS version distinction will be unnecessary once we discontinue support for 10.15.